From e6c131c76f99901db781cde18a75e236e7db51f5 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 13 Jan 2023 09:50:04 +0800
Subject: [PATCH 26/71] i3c: svc: Revert "npcm845: improve i3c performance"

1. reverts commit 0eb6ff5cf8f538cc1504c5369b9b0955cf26546b
   to fix compatibility issue
2. add more debug log
---
 drivers/i3c/master/svc-i3c-master.c | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/drivers/i3c/master/svc-i3c-master.c b/drivers/i3c/master/svc-i3c-master.c
index 1cc3bfe20895..68670ed35793 100644
--- a/drivers/i3c/master/svc-i3c-master.c
+++ b/drivers/i3c/master/svc-i3c-master.c
@@ -601,10 +601,10 @@ static int svc_i3c_master_bus_init(struct i3c_master_controller *m)
 	}
 	pp_high_period = (ppbaud + 1) * fclk_period_ns;
 
-	/* Configure Open-Drain clock, up to 4.17MHz/240ns */
-	odhpp = 1;
+	/* Using I3C Open-Drain mode, target is 1MHz/1000ns with 50% duty cycle */
+	odhpp = 0;
 	high_period_ns = (ppbaud + 1) * fclk_period_ns;
-	odbaud = DIV_ROUND_UP(200, high_period_ns) - 1;
+	odbaud = DIV_ROUND_UP(500, high_period_ns) - 1;
 	od_low_period_ns = (odbaud + 1) * high_period_ns;
 
 	/* Configure for I2C mode */
@@ -1501,11 +1501,18 @@ static int svc_i3c_master_send_ccc_cmd(struct i3c_master_controller *m,
 {
 	struct svc_i3c_master *master = to_svc_i3c_master(m);
 	bool broadcast = cmd->id < 0x80;
+	int ret;
 
 	if (broadcast)
-		return svc_i3c_master_send_bdcast_ccc_cmd(master, cmd);
+		ret = svc_i3c_master_send_bdcast_ccc_cmd(master, cmd);
 	else
-		return svc_i3c_master_send_direct_ccc_cmd(master, cmd);
+		ret = svc_i3c_master_send_direct_ccc_cmd(master, cmd);
+
+	if (ret)
+		dev_dbg(master->dev, "send ccc 0x%02x %s, ret = %d\n",
+				cmd->id, broadcast ? "(broadcast)" : "", ret);
+
+	return ret;
 }
 
 static int svc_i3c_master_priv_xfers(struct i3c_dev_desc *dev,
-- 
2.34.1

